content/blog/2018/egress-mongo/index.md

```diff
@@ -386,6 +386,14 @@ enable Mixer policy enforcement based on that identity. By enabling mutual TLS y
       name: egressgateway-for-mongo
     spec:
       host: istio-egressgateway.istio-system.svc.cluster.local
+      trafficPolicy:
```

Comment:
Since this isn't used anyway, wouldn't it be good enough to make the traffic policy same as default, i.e., just globally set the mode and nothing else?
```
trafficPolicy:
  tls:
    mode: ISTIO_MUTUAL
```
This is much simpler and all you really need to avoid the auth conflict, right?

Reply:
I think the issue is with the subset, which I need to define to set SNI. Once I define a subset without the tls mode, it will cancel the global mode, I am afraid.

Reply:
The issue is the same as with https://github.com/istio/istio/blob/master/samples/bookinfo/networking/destination-rule-all-mtls.yaml vs. https://github.com/istio/istio/blob/master/samples/bookinfo/networking/destination-rule-all.yaml.

When you define a subset, you have to specify the TLS mode.

Reply:
Yes, I understand. I meant you should leave the trafficPolicy in the subsets as they are, but when you add the top level trafficPolicy, you can simplify it. Instead of adding:
```
      trafficPolicy:
        loadBalancer:
          simple: ROUND_ROBIN
        portLevelSettings:
        - port:
            number: 443
          tls:
            mode: ISTIO_MUTUAL
```
you can simplify it to just:
```
      trafficPolicy:
        tls:
          mode: ISTIO_MUTUAL
```

Reply:
But why should I interfere with other ports? Suppose some user would want to use port 80 without mTLS and without TLS origination for some external services, while using port 443 with mTLS. I prefer to be specific here.

Reply:
My point is that if you had not created a DestinationRule at all, the [default destination rule in istio-system](https://github.com/istio/istio/blob/f03880b23dfd1aeef0b734e5766d87bdec3e1c3f/install/kubernetes/helm/istio/charts/security/templates/enable-mesh-mtls.yaml#L23) uses mTLS on all ports by default. So why change that for ports you are not using in the example? If users want to turn off mTLS for some other port, they would add another port level setting for it, rather then relying on the overridden default (which isn't even explicit - it's only overridden because this DestinationRule doesn't declare a top level trafficPolicy).